src/js/modal.js

```diff
@@ -0,0 +1,156 @@
+export function createModal(challenge) {
+
+    const container = document.createElement('div');
+    container.classList.add('modal-container');
+    document.body.appendChild(container);
+
+    renderContent(step1(challenge));
+    
+    return container;
+}
+
+export function removeModal() {
+    document.querySelector('.modal-container').remove();
+}
+
+function renderContent(content) {
+    const container = document.querySelector('.modal-container'); 
+    container.innerHTML = '';
+    container.appendChild(content);
+}
+
+function step1(challenge) {
+
+    const form = document.createElement('form');
+    form.classList.add('content');
+
+    const h1 = document.createElement('h1');
+    h1.textContent = `Book room ${challenge.title} (step 1)`;
+    form.appendChild(h1);
+
+    const p = document.createElement('p');
+    p.textContent = 'What date would you like to come?';
+    form.appendChild(p);
+
+    const label = document.createElement('label');
+    label.textContent = 'Date';
+    label.for = 'input[type=date]';
```

Comment:
The `for` property of a label should contain the ID of the `<input>` that it represents, not a CSS selector.